### PR TITLE
Add scheduled mutation testing to generated repos

### PR DESCRIPTION
**Pull request by leynos**
## Summary

This branch rolls the estate-wide scheduled mutation-testing pattern into the Copier template so newly generated repositories receive the applicable shared workflow by default.

Python projects with a 3.13 or newer baseline receive mutmut configuration and a scheduled mutmut job. Rust-enabled projects receive cargo-mutants independently of the Python baseline. Projects with neither capability do not render a mutation-testing workflow. The default Python baseline moves from 3.10 to 3.12, and malformed baseline answers now fail at the Copier boundary before the hidden mutation-support answer is derived.

## Review walkthrough

- Start with [`copier.yml`](https://github.com/leynos/agent-template-python/blob/60074b85c966d52426386deb005202d97e3d1f4d/copier.yml) for the Python 3.12 default, strict `3.X` validation, and defensive `mutmut_supported` computation.
- Review the [mutation-testing workflow template](https://github.com/leynos/agent-template-python/blob/60074b85c966d52426386deb005202d97e3d1f4d/template/.github/workflows/%7B%25%20if%20mutmut_supported%20or%20use_rust%20%25%7Dmutation-testing.yml%7B%25%20endif%20%25%7D.jinja) for the scheduled shared-workflow callers pinned to `leynos/shared-actions@859416a90eb3987b46a57682c5d6b8964ad3f0a6`.
- Continue with [`template/pyproject.toml.jinja`](https://github.com/leynos/agent-template-python/blob/60074b85c966d52426386deb005202d97e3d1f4d/template/pyproject.toml.jinja) for the gated mutmut configuration.
- Review the generated [users' guide](https://github.com/leynos/agent-template-python/blob/60074b85c966d52426386deb005202d97e3d1f4d/template/docs/users-guide.md.jinja), generated [developers' guide](https://github.com/leynos/agent-template-python/blob/60074b85c966d52426386deb005202d97e3d1f4d/template/docs/developers-guide.md), and template-repository [developers' guide](https://github.com/leynos/agent-template-python/blob/60074b85c966d52426386deb005202d97e3d1f4d/docs/developers-guide.md) for conditional user and maintainer guidance.
- Finish with [`tests/test_template.py`](https://github.com/leynos/agent-template-python/blob/60074b85c966d52426386deb005202d97e3d1f4d/tests/test_template.py) and the shared [rendering helper](https://github.com/leynos/agent-template-python/blob/60074b85c966d52426386deb005202d97e3d1f4d/tests/helpers/rendering.py) for rendering-gate, malformed-input, workflow, TOML and actionlint coverage.

## Gating and scheduling

- The mutmut job and `[tool.mutmut]` section render only for Python 3.13 or newer because the shared workflow helpers require Python 3.13 or newer.
- The cargo-mutants job remains gated independently by `use_rust` and targets `rust_extension/` through `extra-crate-dirs` because generated repositories have no root `Cargo.toml`.
- When both gates are off, Copier omits the workflow file by rendering its conditional filename as empty.
- The generated schedule is `30 9 * * *` (09:30 UTC), with guidance to stagger it per repository.
- Manual cargo dispatch retains a known shared-workflow limitation: full runs create one root-target leg despite the absent root crate. `shard-count: 1` confines the failing root target to one leg while the extension target still runs.

## Validation

- `make check-fmt`: passed; 19 files checked.
- `make lint`: passed; Ruff clean and interrogate coverage 100%.
- `make typecheck`: passed.
- `make test`: passed; 41 passed, 2 skipped and 3 snapshots passed.
- Focused mutation-gating and malformed-version tests: 8 passed.
- Rendered workflow variants YAML-parse and pass actionlint; rendered project metadata TOML-parses with the expected Python and Ruff baselines.
- Rendered Rust variant: `cargo test --manifest-path rust_extension/Cargo.toml` passed with 1 test.

## References

- [Lody session](https://lody.ai/leynos/sessions/5f78fb69-f35a-49da-b7ff-5218dc2abd03)
- [mutmut reusable-workflow guide](https://github.com/leynos/shared-actions/blob/main/docs/mutation-mutmut-workflow.md)
- [cargo-mutants reusable-workflow guide](https://github.com/leynos/shared-actions/blob/main/docs/mutation-cargo-workflow.md)

### CODE REVIEW

File: template/.github/workflows/{% if mutmut_supported or use_rust %}mutation-testing.yml{% endif %}.jinja [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    permissions:
+      contents: read
+      id-token: write # OIDC workflow-source resolution
+    uses: leynos/shared-actions/.github/workflows/mutation-mutmut.yml@2b09d10192627fd6e1034e7c12625dd266b45503
+    with:
+      # Flat package layout: the package directory sits at the
+      # repository root, so no module prefix is stripped.
+      paths: "{{ package_name }}/"
+      module-prefix-strip: ""
+      python-version: "{{ python_version }}"
```
[the remark this answers is not in the document]
Comment (leynos) [reply template/.github/workflows/{% if mutmut_supported or use_rust %}mutation-testing.yml{% endif %}.jinja:42]: @coderabbitai Has this now been resolved in the latest commit?

Use codegraph analysis to determine your answer.

If this comment is now resolved, please mark it as such using the API. Otherwise, please provide an AI agent prompt for the remaining work to be done to address this comment.

File: tests/test_template.py [hunk 2]
```diff
@@ -494,3 +496,97 @@ def test_generated_github_workflows_match_act_validation_contract(
...
+        renders only with the Rust extension, and the workflow file is
+        absent when both gates are off.
+    """
+    project = copier.copy(
+        tmp_path / target_dir,
+        project_name="MutationProj",
+        package_name="mutation_pkg",
+        use_rust=use_rust,
+        python_version=python_version,
+    )
```
[the remark this answers is not in the document]
Comment (leynos) [reply tests/test_template.py]: @coderabbitai Has this now been resolved in the latest commit?

Use codegraph analysis to determine your answer.

If this comment is now resolved, please mark it as such using the API. Otherwise, please provide an AI agent prompt for the remaining work to be done to address this comment.

File: copier.yml [hunk 3]
```diff
@@ -18,5 +18,12 @@ use_rust:
...
-  default: "3.10"
+  default: "3.12"
   help: Minimum supported Python version
+
+# Computed: mutmut mutation testing needs a base interpreter of 3.13 or
+# greater because the shared workflow's helper scripts require >= 3.13.
+mutmut_supported:
+  type: bool
+  default: "{{ python_version.split('.')[1] | int >= 13 }}"
+  when: false
```
[the remark this answers is not in the document]
Comment (leynos) [reply copier.yml:37]: @coderabbitai Has this now been resolved in the latest commit?

Use codegraph analysis to determine your answer.

If this comment is now resolved, please mark it as such using the API. Otherwise, please provide an AI agent prompt for the remaining work to be done to address this comment.

File: template/docs/developers-guide.md [hunk 4]
```diff
@@ -53,6 +53,20 @@ actions under `.github/`.
...
   the CodeScene coverage CLI installer, computes its SHA-256 digest, and writes
   the result to the `CODESCENE_CLI_SHA256` repository variable.
+{% if mutmut_supported or use_rust %}
+- `.github/workflows/mutation-testing.yml` runs daily at 09:30 UTC and supports
+  manual dispatch. It delegates to reusable workflows from `leynos/shared-actions`;
+  stagger the generated cron schedule before adopting it alongside other
+  repositories.
+{% if mutmut_supported %}
+  The mutmut job is generated only when the minimum Python version is 3.13 or
+  newer, because the shared workflow helpers require Python 3.13 or newer.
```
[the remark this answers is not in the document]
Comment (leynos) [reply template/docs/developers-guide.md]: @coderabbitai Has this now been resolved in the latest commit?

Use codegraph analysis to determine your answer.

If this comment is now resolved, please mark it as such using the API. Otherwise, please provide an AI agent prompt for the remaining work to be done to address this comment.

File: tests/helpers/pyproject_contracts.py [hunk 5]
```diff
@@ -20,7 +20,7 @@ def _assert_pyproject_contracts(
     """Assert generated Python packaging contracts."""
     project = require_mapping(pyproject, "project", "pyproject.toml")
     assert project.get("name"), "expected generated project metadata to include a name"
-    assert project.get("requires-python") == ">=3.10", (
+    assert project.get("requires-python") == ">=3.12", (
         "expected generated pyproject.toml to use the requested Python version"
     )
```
[the remark this answers is not in the document]
Comment (leynos) [reply tests/helpers/pyproject_contracts.py]: @coderabbitai Has this now been resolved in the latest commit?

Use codegraph analysis to determine your answer.

If this comment is now resolved, please mark it as such using the API. Otherwise, please provide an AI agent prompt for the remaining work to be done to address this comment.

File: tests/test_helpers.py [hunk 6]
```diff
@@ -41,6 +41,19 @@
 REPO_ROOT = Path(__file__).resolve().parent.parent
 
 
+"""Validate direct helper-module error handling and edge cases.
+The tests in this module exercise support helpers without rendering a full
+Copier project.  They keep helper fallibility contracts explicit by checking
+``pytest.fail`` conversion paths, generated-file schema helpers, and tooling
+contract assertions directly.
+"""
```
[the remark this answers is not in the document]
Comment (leynos) [reply tests/test_helpers.py]: @coderabbitai Has this now been resolved in the latest commit?

Use codegraph analysis to determine your answer.

If this comment is now resolved, please mark it as such using the API. Otherwise, please provide an AI agent prompt for the remaining work to be done to address this comment.
